src/main/java/codechicken/lib/render/CCRenderPipeline.java

```diff
@@ -1,21 +1,37 @@
 package codechicken.lib.render;
 
 import java.util.ArrayList;
+import java.util.Collections;
 
 import codechicken.lib.render.CCRenderState.IVertexOperation;
 import codechicken.lib.render.CCRenderState.VertexAttribute;
 
+@SuppressWarnings("ForLoopReplaceableByForEach")
 public class CCRenderPipeline {
 
+    private final CCRenderState renderState;
+    private final PipelineBuilder builder;
+
+    public CCRenderPipeline(CCRenderState renderState) {
+        this.renderState = renderState;
+        builder = new PipelineBuilder(renderState);
+    }
+
     public class PipelineBuilder {
 
+        private final CCRenderState renderState;
+
+        public PipelineBuilder(CCRenderState renderState) {
```

Comment:
need no arg overload

though yeah I agree probably no one outside is using this constructor directly, but just in case